packages/js/api/src/framework/__tests__/payload-transformer.spec.ts

```diff
@@ -0,0 +1,383 @@
+import {Order, BillingOrderAddress, ShippingOrderAddress, OrderLineItem, OrderShippingLine, OrderFeeLine, OrderTaxRate, OrderCouponLine, MetaData, OrderRefundLine} from "../../models";
+import {createBillingAddressTransformer, createOrderTransformer, createShippingAddressTransformer} from "../../repositories/rest/orders/transformer";
+
+const jsonPayloadFull = JSON.stringify({
+	id: 1218,
+	parent_id: 0,
+	status: 'pending',
+	currency: 'USD',
+	version: '6.1.0',
+	prices_include_tax: false,
+	date_created: '2021-12-01T05:43:38',
+	date_modified: '2021-12-01T07:31:05',
+	discount_total: '5.00',
+	discount_tax: '0.60',
+	shipping_total: '0.00',
+	shipping_tax: '0.00',
+	cart_tax: '6.00',
+	total: '56.00',
+	total_tax: '6.00',
+	customer_id: 0,
+	order_key: 'wc_order_GFHWVmAiamh0B',
+	billing: {
+		first_name: 'Billing First Name',
+		last_name: 'Billing Last Name',
+		company: 'Billing Company',
+		address_1: 'Billing Address 1',
+		address_2: 'Billing Address 2',
+		city: 'Billing City',
+		state: 'Billing State',
+		postcode: 'Billing Postcode',
+		country: 'Billing Country',
+		email: 'Billing Email',
+		phone: 'Billing Phone'
+	},
+	shipping: {
+		first_name: 'Shipping First Name',
+		last_name: 'Shipping Last Name',
+		company: 'Shipping Company',
+		address_1: 'Shipping Address 1',
+		address_2: 'Shipping Address 2',
+		city: 'Shipping City',
+		state: 'Shipping State',
+		postcode: 'Shipping Postcode',
+		country: 'Shipping Country',
+		phone: 'Shipping Phone'
+	},
+	payment_method: 'Foo Payment Method',
+	payment_method_title: 'Foo Payment Method Title',
+	transaction_id: 'Foo Transaction ID',
+	customer_ip_address: 'Foo Customer IP Address',
+	customer_user_agent: 'Foo Customer User Agent',
+	created_via: 'admin',
+	customer_note: 'Foo Customer Note',
+	date_completed: 'Foo Date Completed',
+	date_paid: 'Foo Date Paid',
+	cart_hash: 'Foo Cart Hash',
+	number: '1218',
+	meta_data: [
+		{
+			id: 123,
+			key: "Foo Metadata Key 1",
+			value: "Foo Metadata Value 1"
+		}
+	],
+	line_items: [
+		{
+			id: 6137,
+			name: 'Belt',
+			product_id: 16,
+			variation_id: 0,
+			quantity: 1,
+			tax_class: '',
+			subtotal: '55.00',
+			subtotal_tax: '6.60',
+			total: '50.00',
+			total_tax: '6.00',
+			taxes: [ { id: 1, total: '6', subtotal: '6.6' } ],
+			meta_data: [],
```

Comment:
We probably should add meta data in all the meta data entries but that can be done in a follow up issue/PR.